R/create_protocol.R

```diff
@@ -156,6 +199,47 @@ create_sfp <- function(
                filename)
     close(original_file)
   }
+
+  # docx
+  if (!missing(from_docx)) {
+    convert_docx_to_rmd(
+      from = from_docx,
+      to = book_filename,
+      dir = path_to_protocol,
+      wrap = 80,
+      overwrite = FALSE,
+      verbose = FALSE)
+    # move relevant sections
+    contents <- readLines(con = file.path(path_to_protocol,
+                                          book_filename))
+    is_title <- str_detect(string = contents, pattern = "^(#{1}\\s{1})")
+    title_numbers <- formatC(x = cumsum(is_title),
+                             width = 2, format = "d", flag = "0")
+    filenames <- str_remove(string = tolower(contents[is_title]),
+                            pattern = "^(#{1}\\s{1})")
+    filenames <- str_remove(string = filenames,
+                            pattern = "\\s$")
+    filenames <- str_replace_all(filenames, pattern = "\\s", replacement = "_")
+    filenames <- paste0(unique(title_numbers), "_", filenames, ".Rmd")
+    # delete the empty template chapters
+    file.remove(
+      list.files(
+        path = path_to_protocol,
+        pattern = "^\\d{2}_",
+        full.names = TRUE
+      )
+    )
+    # create new chapters
+    file.create(file.path(path_to_protocol, filenames))
+    # and add chapter contents from docx
+    for (chapter in unique(cumsum(is_title))) {
+      chapter_file <- file.path(path_to_protocol, filenames[chapter])
+      chapter_contents <- contents[chapter == cumsum(is_title)]
+      writeLines(text = chapter_contents,
+                 con = chapter_file)
+    }
```

Comment:
I suggest to delete the original Rmd after this part is finished. I first had cases on which the existence of that Rmd file led to an error (it has the same name as the one produced by bookdown). After running with `debugonce()` this problem stopped for some reason, but anyway removing it seems safer.

Reply:
Strange, this should not happen because of https://github.com/inbo/protocolhelper/pull/9/commits/5f23796ddb26e3a56ef693eb16485a260ad2d240

Reply:
The Rmd is deleted after bookdown compiling indeed. What I mean is: an Rmd with the same name is first created by `convert_docx_to_rmd()`, already before bookdown-compiling. It is not needed for bookdown but can hinder compilation (cf. what you get with bookdown compilation failures: the Rmd must first be deleted before running again). So I suggest deleting it before running `render_book()`.

Reply:
See 
790ed2f